File: data-demo/src/main/java/com/debug/demo/service/impl/UserServiceImpl.java
package com.debug.demo.service.impl;

import com.debug.demo.entity.User;
import com.debug.demo.mapper.UserMapper;
import com.debug.demo.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.sql.SQLException;

/**
 * @author Lyb
 * @since 2020-01-19
 */
@Service
public class UserServiceImpl implements UserService {

    private final UserMapper userMapper;

    @Autowired
    public UserServiceImpl(UserMapper userMapper) {
        this.userMapper = userMapper;
    }

    @Override
    public Mono<Void> save(User user) {
        return Mono.justOrEmpty(userMapper.insert(user))
                .switchIfEmpty(Mono.error(new SQLException("用户信息保存失败")))
                .then();
    }

    @Override
    public Mono<User> query(Long id) {
        return Mono.justOrEmpty(userMapper.selectById(id))
                .switchIfEmpty(Mono.error(new SQLException("用户不存在")));
    }

    @Override
    public Flux<User> list() {
        return Flux.fromIterable(userMapper.selectList(null));
    }
}

File: configuration/src/main/java/com/debug/configuration/service/impl/UserServiceImpl.kt
package com.debug.configuration.service.impl

import com.debug.configuration.constant.ResCode
import com.debug.configuration.entity.User
import com.debug.configuration.repository.UserRepository
import com.debug.configuration.req.LoginParam
import com.debug.configuration.res.Res
import com.debug.configuration.service.UserService
import com.debug.configuration.util.toMD5
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

/**
 * 用户服务
 * @author Lyb
 * @since 2019-11-27
 */
@Service
class UserServiceImpl @Autowired constructor(private val userRepository: UserRepository) : UserService {

    private val logger: Logger = LoggerFactory.getLogger(javaClass);

    override fun checkLogin(loginParam: LoginParam): Res<*> {
        val res: Res<*> = Res<Any>()
        logger.debug("当前登录用户：{}，密码：{}", loginParam.username, loginParam.password.toMD5())
        val user: User? = userRepository.findByUsernameAndPassword(loginParam.username, loginParam.password.toMD5())
        if (user == null) {
            res.msg = "用户名或密码错误"
        } else {
            res.resCode = ResCode.SUCCESS
            res.msg = "登录成功"
        }
        return res
    }
}